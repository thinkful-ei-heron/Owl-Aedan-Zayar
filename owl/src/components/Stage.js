import React from 'react';
import './Stage.css';

export default class Stage extends React.Component {

    render() {
        let filtered = this.props.part.filter(item => item.onStage === true);
        let staged = filtered.map(item => {
            return (
                <div className ='staged-participant' key={item.id}>
                    <div className = 'staged-name'>
                        <span>{item.name}</span>
                        <div className ='stage-icons'/>
                    </div>
                    <div>
                        <img src={item.avatar} alt="Avatar"/>
                    </div>
                </div>
            );
        });
        return (
            <div className ='Stage'>
                {staged}
            </div>
        );
    }

}