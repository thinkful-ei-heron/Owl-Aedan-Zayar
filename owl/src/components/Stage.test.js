import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';
import store from '../store';

describe('Stage component', () => {
    it('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Stage part={store.participants}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Renders as expected', () => {
        const tree = renderer
        .create(<Stage part={store.participants}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});