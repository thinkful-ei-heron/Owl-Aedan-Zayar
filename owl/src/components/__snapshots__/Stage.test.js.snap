// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage component Renders as expected 1`] = `
<div
  className="Stage"
>
  <div
    className="staged-participant"
  >
    <div
      className="staged-name"
    >
      <span>
        Koren Templeton
      </span>
      <div
        className="stage-icons"
      />
    </div>
    <div>
      <img
        alt="Avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
  <div
    className="staged-participant"
  >
    <div
      className="staged-name"
    >
      <span>
        Ashla Attwool
      </span>
      <div
        className="stage-icons"
      />
    </div>
    <div>
      <img
        alt="Avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
</div>
`;
